source/common/router/upstream_request.cc

```diff
@@ -734,14 +721,6 @@ void UpstreamRequest::readDisableOrDefer(bool disable) {
 
 void UpstreamRequest::DownstreamWatermarkManager::onAboveWriteBufferHighWatermark() {
   ASSERT(parent_.upstream_);
-
-  // There are two states we should get this callback in: 1) the watermark was
-  // hit due to writes from a different filter instance over a shared
-  // downstream connection, or 2) the watermark was hit due to THIS filter
-  // instance writing back the "winning" upstream request. In either case we
-  // can disable reads from upstream.
-  ASSERT(!parent_.parent_.finalUpstreamRequest() ||
```

Comment:
What is the reason to get rid of this and two similar assertions below? I see that `finalUpstreamRequest()` and `upstreamRequests()` are no longer part of router_filter_interface, is it because these assertions are no longer needed?